### PR TITLE
Expand Real Image Datasets

### PR DESCRIPTION
**Pull request by dylanuys**
- Adding the following datasets to increase diversity of validator real image challenges:
  - bitmind/MS-COCO-unique-256
  - bitmind/AFHQ
  - bitmind/lfw
  - bitmind/caltech-256
  - bitmind/caltech-101
  - "bitmind/dtd
- Adding uniform sampling across all real image datasets
- Randomizing whether or not to apply augmentations to add diversity to synthetic images

### CODE REVIEW

File: bitmind/validator/forward.py [hunk 1]
```diff
@@ -146,7 +146,7 @@ async def forward(self):
         performance_tracker=self.performance_tracker)
     
     # Logging image source and verification details
-    source_name = wandb_data['model'] if 'model' in wandb_data else wandb_data['dataset']
+    source_name = wandb_data['model'] if 'model' in wandb_data else wandb_data['source_dataset']
```
Comment (aliang322) [inline bitmind/validator/forward.py:153]: Q: When would wandb_data contain 'model'? Could use a comment explaining the logic here and why we want source_dataset in some cases

**Review by aliang322 — COMMENTED** [review]
Added a minor comment about wandb logging. Otherwise looks good!

File: bitmind/validator/forward.py [hunk 2]
```diff
@@ -60,15 +78,11 @@ async def forward(self):
...
-
-        wandb_data['dataset'] = source_dataset
-        wandb_data['image_index'] = idx[0]
-        wandb_data['image_name'] = sample['id']
-
+        random_idx = np.random.randint(0, self.total_real_images)
+        source_dataset, local_index = sample_random_real_image(self.real_image_datasets, self.total_real_images)
+        wandb_data['source_dataset'] = source_dataset.huggingface_dataset_name
+        wandb_data['source_image_index'] = local_index
+        sample = source_dataset[local_index]
```
Comment (aliang322) [inline bitmind/validator/forward.py:84]: Q: Why are we not retrying for real images v.s. retrying for fake images?

File: bitmind/validator/forward.py [hunk 3]
```diff
@@ -33,6 +33,24 @@
 from bitmind.image_transforms import random_aug_transforms
 
 
+def sample_random_real_image(datasets, total_images, retries=10):
+    random_idx = np.random.randint(0, total_images)
+    source, idx = sample_real_image(datasets, random_idx)
+    if source[idx]['image'] is None:
```
Comment (aliang322) [inline bitmind/validator/forward.py:39]: Q: Likely not an issue in practice but just sanity checking, it should be highly unlikely that the recursion here would continue for an extended period due to many sampled images repeated == None, right?

File: bitmind/validator/forward.py (hunk 2 above)
Comment (dylanuys) [reply bitmind/validator/forward.py:84]: retries are built into sample_random_real_image, but for synthetic images there's an additional retry mechanism for failed generation

File: bitmind/validator/forward.py (hunk 1 above)
Comment (dylanuys) [reply bitmind/validator/forward.py:153]: Will update with a comment explaining the logic!

**Review by kenobijon — APPROVED** [review]
LGTM

**Pull request merged**